cardano-api/src/Cardano/Api.hs

```diff
@@ -164,19 +264,29 @@ buildByronTransaction ins outs =
     bTxHash = coerce $ Crypto.hashRaw (LBS.fromStrict bTxCbor)
 
 
-buildShelleyTransaction :: [TxIn] -> [TxOut] -> SlotNo -> Lovelace -> TxUnsigned
-buildShelleyTransaction txins txouts ttl fee =
-    TxUnsignedShelley $
-      Shelley.TxBody
-        (Set.fromList (map toShelleyTxIn  txins))
-        (Seq.fromList (map toShelleyTxOut txouts))
-        Seq.empty                -- certificates
-        (Shelley.Wdrl Map.empty) -- withdrawals
-        (toShelleyLovelace fee)
-        ttl
-        Shelley.SNothing         -- update proposals
-        Shelley.SNothing         -- metadata hash
-
+buildShelleyTransaction
+  :: [TxIn]
+  -> [TxOut]
+  -> SlotNo
+  -> Lovelace
+  -> [Certificate]
+  -> TxUnsigned
+buildShelleyTransaction txins txouts ttl fee certs = do
+  let relevantCerts = catMaybes [ certDiscrim c | c <- certs ]
```

Comment:
A little confused about this usage of `catMaybes`. `certDiscrim` doesn't appear to ever return `Nothing`, so couldn't it instead be:
```
   certDiscrim :: Certificate -> ShelleyCertificate
   certDiscrim (ShelleyDelegationCertificate delegCert) = delegCert
   certDiscrim (ShelleyStakePoolCertificate sPoolCert) = sPoolCert
```

and then this would be:
```
  let relevantCerts = map certDiscrim certs
```

Reply:
Yep this slipped through due to rebasing/cherry picking. Good catch.